Jenkinsfile

```diff
@@ -125,22 +121,136 @@ pipeline {
             }
           }
+
+          post {
+            always {
+              archiveArtifacts "results/**/*"
+              junit "results/results.xml"
+              withCredentials([
+                [
+                  $class: "AmazonWebServicesCredentialsBinding",
+                  credentialsId: "f40eebe0-f9aa-4336-b460-b2c4d7876fde",
+                  accessKeyVariable: "AWS_ACCESS_KEY_ID",
+                  secretKeyVariable: "AWS_SECRET_ACCESS_KEY"
+                ]
+              ]) {
+                sh "./system-tests/_scripts/delete-cluster.sh"
+              }
+            }
+          }
+        }
+      }
+    }
+
+    stage("System Test previous versions") {
+      when {
+        expression {
+          master_branches.contains(BRANCH_NAME)
         }
       }
+      steps {
+        stage("1.13 nightly") {
+          steps {
+            withCredentials([
+              [
+                $class: "AmazonWebServicesCredentialsBinding",
+                credentialsId: "f40eebe0-f9aa-4336-b460-b2c4d7876fde",
+                accessKeyVariable: "AWS_ACCESS_KEY_ID",
+                secretKeyVariable: "AWS_SECRET_ACCESS_KEY"
+              ]
+            ]) {
+              sh '''
+                INSTALLER_URL="https://downloads.dcos.io/dcos/testing/1.13/dcos_generate_config.sh" ./system-tests/_scripts/launch-cluster.sh
+                export CLUSTER_URL=\$(cat /tmp/cluster_url.txt)
+                export CLUSTER_AUTH_TOKEN=\$(./system-tests/_scripts/get_cluster_auth.sh)
+                export CLUSTER_AUTH_INFO=\$(echo '{ "uid": "albert@bekstil.net", "description": "albert" }' | base64)
+                DCOS_CLUSTER_SETUP_ACS_TOKEN="\$CLUSTER_AUTH_TOKEN" dcos cluster setup "\$CLUSTER_URL" --provider=dcos-oidc-auth0 --insecure
+                npm run test:system
+              '''
+            }
+          }
+
+          post {
+            always {
+              archiveArtifacts "results/**/*"
+              junit "results/results.xml"
+              withCredentials([
+                [
+                  $class: "AmazonWebServicesCredentialsBinding",
+                  credentialsId: "f40eebe0-f9aa-4336-b460-b2c4d7876fde",
+                  accessKeyVariable: "AWS_ACCESS_KEY_ID",
+                  secretKeyVariable: "AWS_SECRET_ACCESS_KEY"
+                ]
+              ]) {
+                sh "./system-tests/_scripts/delete-cluster.sh"
+              }
+            }
+          }
+        }
 
-      post {
-        always {
-          archiveArtifacts "results/**/*"
-          junit "results/results.xml"
-          withCredentials([
-            [
-              $class: "AmazonWebServicesCredentialsBinding",
-              credentialsId: "f40eebe0-f9aa-4336-b460-b2c4d7876fde",
-              accessKeyVariable: "AWS_ACCESS_KEY_ID",
-              secretKeyVariable: "AWS_SECRET_ACCESS_KEY"
-            ]
-          ]) {
-            sh "./system-tests/_scripts/delete-cluster.sh"
+        // stage("1.13.0") {
```

Comment:
we need this once 1.13.0 is GA to make sure independent updates work with 1.13.0